packages/@o3r/schematics/src/utility/wrapper.ts

```diff
@@ -0,0 +1,36 @@
+import type { Rule } from '@angular-devkit/schematics';
+import { performance } from 'node:perf_hooks';
+import { getEnvironmentInfo } from './environment';
+
+export const createSchematics = <S>(
+  schematicsFn: (opts: S) => Rule
+): ((opts: S) => Rule) => (options: S) => (tree, context) => {
+    const startTime = performance.now();
+    let error;
+    try {
+      return schematicsFn(options)(tree, context);
```

Comment:
In case this function is returning a `promise` (part of the `Rule` definition)  does the `try catch` still working?
Maybe safer to `await schematicsFn` before returning

Reply:
It should work fine I added a test case to be sure

we can not just `await` a `Rule` because it could be an `Observable` or a `Promise`

Reply:
then
```typescript
import { isObservable } from "rxjs";
await (isObservable(result) ? lastValueFrom(result): result)
```

Reply:
As I said, I added a test when the schematic throw an error and it seems everything is fine

Reply:
Get it, fixed it and add a test case :)